notebooks/Defining-a-Heuristic-Policy.ipynb

```diff
@@ -0,0 +1,230 @@
+    "\n",
+    "[1] Kochenderfer, Mykel J. Decision Making Under Uncertainty: Theory and Application. MIT Press, 2015"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "┌ Info: Recompiling stale cache file /home/shushman/.julia/compiled/v1.0/POMDPs/GAotg.ji for POMDPs [a93abf59-7444-517b-a68a-c42f96afdd7d]\n",
+      "└ @ Base loading.jl:1184\n",
+      "┌ Info: Recompiling stale cache file /home/shushman/.julia/compiled/v1.0/POMDPPolicies/2WG4l.ji for POMDPPolicies [182e52fb-cfd0-5e46-8c26-fd0667c990f4]\n",
+      "└ @ Base loading.jl:1184\n",
+      "┌ Info: Recompiling stale cache file /home/shushman/.julia/compiled/v1.0/POMDPModels/GHWgR.ji for POMDPModels [355abbd5-f08e-5560-ac9e-8b5f2592a0ca]\n",
+      "└ @ Base loading.jl:1184\n",
+      "WARNING: Method definition GridWorld(Any...) in module POMDPModels at deprecated.jl:29 overwritten at deprecated.jl:53.\n"
+     ]
+    }
+   ],
+   "source": [
+    "using POMDPs\n",
+    "using POMDPPolicies # For defining a policy\n",
+    "using POMDPModels # For the TigerPOMDP Model\n",
+    "using BeliefUpdaters # To use DiscreteUpdater"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We will define a HeuristicPolicy type that only requires the POMDP instance and the set of valid actions."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "struct HeuristicPolicy{P<:POMDP, A} <: Policy\n",
+    "    pomdp::P\n",
+    "    action_map::Vector{A}\n",
+    "end"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "HeuristicPolicy"
+      ]
+     },
+     "execution_count": 3,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "function HeuristicPolicy(pomdp::POMDP)\n",
+    "    HeuristicPolicy(pomdp, actions(pomdp))\n",
+    "end"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Overriding the POMDPs.jl action function\n",
+    "Now we will define the `action` function, which specifies the behavior of the heuristic policy. It requires the belief state to be represented as a `DiscreteBelief`, i.e. a Probability Mass Function over individual states. It computes the expected single-step reward for each action, given the current belief state, and chooses the maximum one. Note that we must use `POMDPs.action` to override the `action` method of `POMDPs.jl`"
```

Comment:
More precise wording regarding "function" and "method", and I think it's good to call this a greedy policy
```suggestion
    "Now we will define a new method for the `action` function, which specifies the behavior of the heuristic policy. It requires the belief state to be represented as a `DiscreteBelief`, i.e. a Probability Mass Function over individual states. It computes the expected single-step reward for each action, given the current belief state, and chooses the maximum one. This is sometimes called a "greedy" or "myopic" policy. Note that we must use `POMDPs.action` to add a method to the `action` function of `POMDPs.jl`"
```